pkg/api/v1/schema.go

```diff
@@ -336,6 +345,14 @@ type VolumeSource struct {
 	// the process of populating that PVC with a disk image.
 	// +optional
 	DataVolume *DataVolumeSource `json:"dataVolume,omitempty"`
+	// ConfigMapSource represents a reference to a ConfigMap in the same namespace.
+	// More info: https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/
+	// +optional
+	ConfigMap *ConfigMapSource `json:"configMap,omitempty"`
+	// SecretVolumeSource represents a reference to a secret data in the same namespace.
+	// More info: https://kubernetes.io/docs/concepts/configuration/secret/
+	// +optional
+	Secret *v1.SecretVolumeSource `json:"secret,omitempty"`
```

Comment:
Same like above. We can almost use the `SecretVolumeSource`. I don't think that we can pass through permissions right now, so `SecretVolumeSource` and `ConfigMapVolumeSource` should look exactly like on the pod except for the `defaultMode` field.